### PR TITLE
Implemented UserMapper using MapStruct

### PR DESCRIPTION
**Pull request by anavduo**
I have check all the endpoints from User  Controller and they work

### CODE REVIEW

**Comment by DanteIturri** [comment]
Perfect Anna! I don't see any conflict

File: src/main/java/com/alkemy/ong/service/UserService.java [hunk 1]
```diff
@@ -1,18 +1,65 @@
...
-    void deleteUserById(String id);
+    public UserService() {
+    }
+    //Soft deletion method for the User
 
-    UserEntity updateUser(UserEntity entity) ;
+    public void deleteUserById(String id) throws Exception {
+        userRepository.deleteById(id);
+    }
+    //Get all Users from Database.
```
Comment (JonathanBolatti) [inline src/main/java/com/alkemy/ong/service/UserService.java:28]: The service to get a list of users is correct

File: src/main/java/com/alkemy/ong/mapper/UserMapper.java [hunk 2]
```diff
@@ -1,61 +1,17 @@
...
-        return dto;
-
-    }
-}
+@Mapper (componentModel = "spring")
+public interface UserMapper {
+       UserMapper userMapper = Mappers.getMapper(UserMapper.class);
+       UserDto userEntityToUserDto(UserEntity userEntity);
+       UserEntity userDtoToUserEntity(UserDto userDto);
+       List<UserDto> listUserEntityToListUserDto(List<UserEntity>list);
```
Comment (JonathanBolatti) [inline src/main/java/com/alkemy/ong/mapper/UserMapper.java:16]: the implementation of mapstruct to obtain a list of users is correct.

File: src/main/java/com/alkemy/ong/controller/UserController.java [hunk 3]
```diff
@@ -25,36 +22,32 @@
     @Autowired
     UserService userService;
 
-    @Autowired
-    private UserRepository userRepository;
 
-    //method to get a list of users
-    @PreAuthorize("hasAnyRole('ROLE_ADMIN')")
+    //Method to get a list of users
     @GetMapping("/list")
```
Comment (JonathanBolatti) [inline src/main/java/com/alkemy/ong/controller/UserController.java:27]: The controller to get a list of users is correct.

Comment (JonathanBolatti) [inline src/main/java/com/alkemy/ong/controller/UserController.java:27]: The controller to get a list of users is correct.

File: src/main/java/com/alkemy/ong/service/UserService.java (hunk 1 above)
Comment (anavduo) [reply src/main/java/com/alkemy/ong/service/UserService.java:28]: thanks for reading

**Comment by anavduo** [comment]
thanks for your comments!

**Pull request merged**